components/Navigator/components/NavButton/index.jsx

```diff
@@ -119,6 +118,10 @@ class NavButton extends Component {
    * @return {JSX} The icon.
    */
   renderIcon() {
+    if (this.props.path === INDEX_PATH) {
```

Comment:
I see the issue in the tabbar itself, not the navbar: on every click it opens a link which adds one more entry to the history and additionally does several more operations like pipeline requests.
If the tabbar would work correctly the dirty fix here wouldn't be needed.
I think this happens for every tabbar action.

Reply:
Still needs to be done for the other tab-bar actions (no pipeline requests done though, but unnecessary history updates).

This line here can be removed.

Reply:
ok we'll leave it .. doesn't hurt